miner/worker.go

```diff
@@ -1105,27 +1084,29 @@ func (w *worker) commitWork(interruptCh chan int32, timestamp int64) {
 	// validator can try several times to get the most profitable block,
 	// as long as the timestamp is not reached.
 	workList := make([]*environment, 0, 10)
-	var bestWork *environment
+	var prevWork *environment
 	// workList clean up
 	defer func() {
-		for _, w := range workList {
+		for _, wk := range workList {
 			// only keep the best work, discard others.
-			if w == bestWork {
+			if wk == w.current {
 				continue
 			}
-			w.discard()
+			wk.discard()
```

Comment:
only one `triePrefetcher` be reused and `transfered`, this `range` may not necessary

Reply:
the `discard()` logic could be changed, although it only stop triePrefecth right now, it could do something later.